x-pack/plugins/fleet/server/services/agent_policy.ts

```diff
@@ -1080,6 +1091,144 @@ class AgentPolicyService {
     };
   }
 
+  public async createAgentlessAgent(
```

Comment:
Could we move this logic to a separate file with its tests? The logic here is already 1k+ lines long.

Reply:
I've just moved the functionality and created a new service.

- https://github.com/elastic/kibana/pull/187488/commits/d6d6a8a5c4eef099b0d42dfb15067ac335d631c2